File: Java/Medium/Accepted/minimum-numbers-of-function-calls-to-make-target-array_7.java
class Solution {
    public int minOperations(int[] nums) {
        
        int max = 0;
        int count = 0;
        for(int num: nums){
            int temp = 0;
            while(num > 0){
                if(num % 2 == 1){
                    num--;
                    count++;
                }else{
                    num /= 2;
                    temp++;
                }
            }
            max = Math.max(max, temp);
        }
        return count + max;
    }
}

File: C++/Medium/WrongAnswer/minimum-numbers-of-function-calls-to-make-target-array_7.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        
        int ans = 0;
        int maxv = 0;
        for (int num : nums) {
            int zeros = 0;
            while (num) {
                if (num & 1) ans++;
                num >>= 1;
                zeros++;
            }
            maxv = max(maxv, zeros);
        }
        return ans + maxv - 1;
    }
};